### DISCUSSION TITLE
Time-dependent scan_interval

### DISCUSSION THREAD

**Discussion by linux-guru-com** [Chouffy/home_assistant_tgtg#15] [Ideas]
I (at least personally) would have an use-case to have higher or lower scan intervals at certain times of the day.
Idea behind it is, that most of (my) tgtg offers will be updated only once a day within a certain time period (here from 8 pm to 8:20 pm). That update does not happen at exactly the same time on each day. 
Getting alerted once it is available would be extremely helpful to snatch a good deal.
Toi prevent hammering the tgtg server(s) with useless requests, my idea would be, to define a default scan_interval of e.g. 900 s and a elevated scan_interval of e.g. 30 s which only will be considered if we are in a defined time window (e.g. "elevation_window: 2000-2030" maybe even with a list of windows).

What do you think about this idea?

**Comment by Chouffy** [comment]
It would be really great and I would like to do it, but I lack the skill & time to implement that myself.
The `scan_interval` property come from Home Assistant itself and manage the refresh time of the module. I didn't implemented any logic in the module itself.
Maybe it's possible to do an automation that manage a boolean when you want the "high refresh window" to be active and then link different scan_interval or even switch the integration on and off based on this? I honestly don't know if it's even possible in Home Assistant 😅
Be aware that other peoples reported issues when they did too many request, so you may want to limit your pull rate

**Comment by platuszkiewicz** [comment]
I think it would by highly disired feature. home_assistant_tgtg works good, but we are making use of it only when we actually want to order package from TGTG. 
1. I want to order package from TGTG
2. I start sensor with inverval 30 seconds.
3. When package is available, I get notification
4. After order, I turn off the sensor.

In other words, it would be nice to have posibility to turn off/on your sensor by switch on the Home Assistant dashboard.

**Comment by OhaiHFO** [comment]
Perhaps this could be provided by adding a "service" to the integration (https://developers.home-assistant.io/docs/dev_101_services)

Looking at the source code for this integration, you could expose your update function (https://github.com/Chouffy/home_assistant_tgtg/blob/1292f75bad99b9d0dbdf4a48c9b6fc9b928a4176/custom_components/tgtg/sensor.py#L181) as a service.

The service could be used inside of automations and scripts.

This could be used, for instance, to allow end users to set up time-based automations that define periods of time in which the update call happens more frequently that the standard `scan_interval`.

Thank you for you work on this plugin btw, I was pleasantly surprised to come across it 🙏

**Reply by Chouffy** [reply]
Oh wow I didn't knew that, thanks! I'll try to take a look at it, but at the moment it's a bit busy.
Of course I'm more than happy to accept Pull requests on this 😊

**Comment by Butanding** [comment]
Has anybody made progress on this?
I am getting blocked behind captcha if i monitor too much, so it would be nice to reduce the cycle.
it would be nice to have the following config:
A) Specify time based interval
B) Specify item-id based interval (some Items are available usually 30 minutes, some only 2 minutes)
C) A comibination of A) and B)

Or maybe somebody knows a workaround for getting blocked with too much traffic?
Thanks!

**Comment by Guillaumer2h** [comment]
Hi there, this integration is really great, it gave me great hopes but now I'm blocked behind the captcha too. Having a system to decrease the update frequency as described above would be amazing !
For now I'll decrease the update frequency and see if it goes better, but items are going super fast in my region, it might not be as useful anymore.
